alpaca/data/historical/corporate_actions.py

```diff
@@ -0,0 +1,140 @@
+from collections import defaultdict
+from typing import Optional, Union
+
+from alpaca.common.enums import BaseURL
+from alpaca.common.rest import RESTClient
+from alpaca.common.types import RawData
+from alpaca.data.historical.utils import get_data_from_response
+from alpaca.data.models.corporate_actions import CorporateActionsSet
+from alpaca.data.requests import CorporateActionsRequest
+
+
+class CorporateActionsClient(RESTClient):
+    """
+    The REST client for interacting with Alpaca Corporate Actions API endpoints.
+
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        secret_key: Optional[str] = None,
+        oauth_token: Optional[str] = None,
+        use_basic_auth: bool = False,
+        raw_data: bool = False,
+        url_override: Optional[str] = None,
+    ) -> None:
+        """
+        Instantiates a Historical Data Client.
+
+        Args:
+            api_key (Optional[str], optional): Alpaca API key. Defaults to None.
+            secret_key (Optional[str], optional): Alpaca API secret key. Defaults to None.
+            oauth_token (Optional[str]): The oauth token if authenticating via OAuth. Defaults to None.
+            use_basic_auth (bool, optional): If true, API requests will use basic authorization headers.
+            raw_data (bool, optional): If true, API responses will not be wrapped and raw responses will be returned from
+              methods. Defaults to False. This has not been implemented yet.
+            url_override (Optional[str], optional): If specified allows you to override the base url the client points
+              to for proxy/testing.
+        """
+        super().__init__(
+            api_key=api_key,
+            secret_key=secret_key,
+            oauth_token=oauth_token,
+            use_basic_auth=use_basic_auth,
+            api_version="v1beta1",
+            base_url=url_override if url_override is not None else BaseURL.DATA,
+            sandbox=False,
+            raw_data=raw_data,
+        )
+
+    def get_corporate_actions(
+        self, request_params: CorporateActionsRequest
+    ) -> Union[RawData, CorporateActionsSet]:
+        """Returns corporate actions data
+
+        Args:
+            request_params (CorporateActionsRequest): The request params to filter the corporate actions data
+        """
+        params = request_params.to_request_fields()
+
+        if request_params.symbols is not None and len(request_params.symbols) > 0:
```

Comment:
[minor] No need for the is not None check if len > 0. Same for types.

Reply:
right. Thanks!